context/context.go

```diff
@@ -165,7 +165,10 @@ func (c *fsContext) Config() (config.Config, error) {
 	if c.config == nil {
 		cfg, err := config.ParseDefaultConfig()
 		if errors.Is(err, os.ErrNotExist) {
-			cfg = config.NewBlankConfig()
+			cfg, err = config.InitDefaultConfig()
```

Comment:
We've moved away from `Config()` accessor resulting in side-effects such as kicking off authentication flow or making filesystem changes. It would be ideal if it could stay like that (i.e. not result in any mutations)

Instead, could we write the default config in case `config.Write()` was called, but `config.yml` was non-existent?

Reply:
that is much better and i don't know why i thought i couldn't do that